mesa/testProton.py

```diff
@@ -55,17 +55,16 @@ def test_generate_households():
         other_members = set([x for x in test_family if x != member])
         assert other_members == member.neighbors["household"]
     #3
-    for test_simple_family in m.families[:3]:
-        if len(test_simple_family) >= 3:
-            break
-    assert set(test_simple_family[1:]) == test_simple_family[0].neighbors["household"]
-    assert test_simple_family[-1] in test_simple_family[0].neighbors["offspring"]
-    assert test_simple_family[-1] in test_simple_family[1].neighbors["offspring"]
-    if type(test_simple_family[2:]) == list:
-        for son in test_simple_family[2:]:
-            assert son.neighbors["parent"] == set(test_simple_family[:2])
-    else:
-        assert test_simple_family[-1].neighbors["parent"] == set(test_simple_family[:2])
+    for test_simple_family in m.families[:5]:
+        if len(test_simple_family) >= 3 and test_family[0].neighbors["partner"] == test_family[1] and test_family[1].neighbors["partner"]  == test_family[0]:
+            assert set(test_simple_family[1:]) == test_simple_family[0].neighbors["household"]
+            assert test_simple_family[-1] in test_simple_family[0].neighbors["offspring"]
+            assert test_simple_family[-1] in test_simple_family[1].neighbors["offspring"]
+            if type(test_simple_family[2:]) == list:
```

Comment:
What else should it be? Shouldn't we get an error here if it's not a list? Are you worried it could be empty?

What if we replace all of 63-69 with just this:
```
              for son in test_simple_family[2:]:
                    assert son.neighbors["parent"] == set(test_simple_family[:2])
```

Reply:
You're right. I thought families here could be made up of also two elements. But we already have a check here at line 59.